assets/js/components/common/PacketGraph.jsx

```diff
@@ -83,17 +87,13 @@ class PacketGraph extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (!this.props.data || !prevProps.data) return
+    const { recentEvents } = this.props.data
 
-    if (
-      (this.props.data[0] && !prevProps.data[0]) ||
-      (this.props.data[0] && this.state.data.datasets.length == 0) ||
-      (this.props.data[0] && prevProps.data[0] && this.props.data[this.props.data.length-1].id !== prevProps.data[prevProps.data.length-1].id)
-    ) {
+    if (!prevProps.data.recentEvents && recentEvents) {
       clearInterval(this.chartUpdateInterval)
-      this.updateChart(this.props.data)
+      this.updateChart(recentEvents)
       this.chartUpdateInterval = setInterval(() => {
-        this.updateChart(this.props.data)
+        this.updateChart(recentEvents)
```

Comment:
is this just polling that endpoint? could we use subscriptions here so we hit the gql endpoint once, grab all recent events, and then when new ones come in, add to the graph?

Reply:
updateChart just transforms props to state for the chart Component, it doesn't hit the endpoint.